data-prepper-expression/src/test/java/org/opensearch/dataprepper/expression/GenericExpressionEvaluator_ArithmeticIT.java

```diff
@@ -109,20 +111,95 @@ void testArithmeticExpressionEvaluatorWithMultipleThreads(final String expressio
     @ParameterizedTest
     @MethodSource("invalidExpressionArguments")
     void testArithmeticExpressionEvaluatorInvalidInput(final String expression, final Event event, final Class expectedClass) {
+
         final GenericExpressionEvaluator evaluator = applicationContext.getBean(GenericExpressionEvaluator.class);
-        Object result = evaluator.evaluate(expression, event);
-        assertThat(result, not(instanceOf(expectedClass)));
+        if (expectedClass == null) {
+            assertThrows(ExpressionEvaluationException.class, () -> evaluator.evaluate(expression, event));
+        } else {
+            Object result = evaluator.evaluate(expression, event);
+            assertThat(result, not(instanceOf(expectedClass)));
+        }
```

Comment:
We should separate out these tests. These are two separate behaviors that should be handled separately.